localstack/services/stepfunctions/asl/component/state/state_execution/state_map/item_reader/reader_config/reader_config_props.py

```diff
@@ -0,0 +1,23 @@
+from typing import Any, Final
+
+from localstack.services.stepfunctions.asl.component.state.state_execution.state_map.item_reader.reader_config.max_items_decl import (
+    MaxItemsDecl,
+)
+from localstack.services.stepfunctions.asl.parse.typed_props import TypedProps
+
+
+class ReaderConfigProps(TypedProps):
+    _UNIQUE_SUB_INSTANCES: Final[set[type]] = {MaxItemsDecl}
```

Comment:
that's meta with `type` being the type of the instance 🤯

Reply:
Indeed, ASL Component and EvaluationComponents are stored and retrieved via type through TypedProps objects (here is an example https://github.com/localstack/localstack/blob/master/localstack/services/stepfunctions/asl/parse/preprocessor.py#L592). This is a strategy to detect duplicate or conflicting language productions during preprocessing.